routes/room/room.js

```diff
@@ -1,22 +1,44 @@
 const express = require('express');
 const { Room } = require('../../models/room');
 const validateRoom = require('./roomValidation');
-
-
+const mongoose = require('mongoose'); 
+const { Booking } = require('../../models/booking');
 
 const router = express.Router();
 
 router.get('/', async (req, res) => {
     try {
-        const room = await Room.find({});
-        res.send(room);
+        const { arrivalDate, departureDate } = req.query;
+
+        if(!arrivalDate && !departureDate){
+            const room = await Room.find({});
+            return res.send(room);
+        }
+        const arrival = new Date(arrivalDate);
+        const departure = new Date(departureDate);
+ 
+        const bookedRooms = await Booking.find({
+            $or: [
+                {
+                    arrivalDate: { $lte: departure,$gte: arrival  },
+                    departureDate: { $gte: arrival, $lte: departure }
+                }
+            ]
+        }).select("roomId");
+       
+        const bookedRoomIds = bookedRooms.map(booking => booking.roomId);
+
+        const availableRooms = await Room.find({
+            _id: { $nin: bookedRoomIds }
+        });
+        if (availableRooms.length === 0) return res.status(404).send('No available rooms for this date');
```

Comment:
Please make sure the message returned to the user is clear by being precise in it. I would suggest some thing more accurate like: "There are no available rooms in the provided dates".